Sprint-3/alarmclock/alarmclock.js

```diff
@@ -1,4 +1,43 @@
-function setAlarm() {}
+function setAlarm() {
+  const inputField = document.getElementById("alarmSet");
+  let timeRemaining = parseInt(inputField.value, 10);
+
+  if (isNaN(timeRemaining) || timeRemaining <= 0) {
+    alert("Please enter a valid positive number!");
+    return;
+  }
+
+  const timeDisplay = document.getElementById("timeRemaining");
+  let interval;
+
+  function updateDisplay() {
+    const minutes = Math.floor(timeRemaining / 60);
+    const seconds = timeRemaining % 60;
+    timeDisplay.innerText = `Time Remaining: ${minutes.toString().padStart(2, "0")}:${seconds.toString().padStart(2, "0")}`;
+  }
+
+  function countdown() {
+    if (timeRemaining > 0) {
+      timeRemaining--;
+      updateDisplay();
+    } else {
+      clearInterval(interval);
+      timeDisplay.style.backgroundColor = "red";
+      playAlarm();
+    }
+  }
+
+  updateDisplay();
+  interval = setInterval(countdown, 1000);
```

Comment:
why 1000? even if it is enough for most cases, if the input is 10 seconds for example, can you save the extra 9900 calls somehow?